app/src/matrix_transform.c

```diff
@@ -33,7 +34,12 @@ uint32_t zmk_matrix_transform_row_column_to_position(uint32_t row, uint32_t colu
     matrix_index = (row * ZMK_MATRIX_COLS) + column;
 
 #ifdef ZMK_KEYMAP_TRANSFORM_NODE
-    return transform[matrix_index];
+    if (matrix_index >= ARRAY_SIZE(transform))
+        return -1;
+    value = transform[matrix_index];
+    if (!value)
+        return -1;
+    return value - 1;
```

Comment:
This could be simplified to
```c
return transform[matrix_index] - 1;
```

Reply:
I defined macro for the offset, and I changed the return value to -EINVAL when the position is not found, and this now seems more correct, so as it is it can't be simplified anymore.